docs/Options.md

```diff
@@ -225,12 +225,13 @@ Mercurial status indicators is shown only when you have dirty repository.
 
 ### Package version (`package`)
 
-> Works for [npm](https://www.npmjs.com/) and [cargo](https://crates.io/) at the moment. Please, help us improve this section!
+> Works for [npm](https://www.npmjs.com/), [cargo](https://crates.io/) and [python](https://www.python.org/) at the moment. Please, help us improve this section!
 
 Package version is shown when repository is a package.
 
 * **npm** — `npm` package contains a `package.json` file. We use `jq`, `python` to parse package version for improving performance and `node` as a fallback. Install [jq](https://stedolan.github.io/jq/) for **improved performance** of this section ([Why?](./Troubleshooting.md#why-is-my-prompt-slow))
 * **cargo** — `cargo` package contains a `Cargo.toml` file. Currently, we use `cargo pkgid`, it depends on `Cargo.lock`. So if package version isn't shown, you may need to run some command like `cargo build` which can generate `Cargo.lock` file.
+* **python** — `python` package contains a `setup.py` or `pyproject.toml` file (either in current dir, or git repo root). Currently, we use [pkg_resources](https://setuptools.readthedocs.io/en/latest/pkg_resources.html) to figure out the package version by package name. The package name is defined using either the [basename](https://docs.python.org/2/library/os.path.html#os.path.basename) of the git repository root directory (in case the current directory is inside a git repo) or the [basename](https://docs.python.org/2/library/os.path.html#os.path.basename) of the current directory.
```

Comment:
There's several bad assumptions here:
1) In the modern years, it is expected to use `importlib-metadata` (present in stdlib) or its compatibility backport `importlib_metadata` (there's a chance something's already installed it from PyPI). `pkg_resources` is deprecated and it's use is highly discouraged.
2) Because `pkg_resources` is deprecated and also due to it being a part of `setuptools`. It may simply be absent from a given env. It is possible to create virtualenvs without setuptools provisioned in them. Also, many downstreams (hello, ubuntu/debian) tend to unvendor bits of the interpreter and when they put setuptools in a separate package, it may not be pulled in by other packages yet — so there's no guarantee of its presence even in the system install.
3) Another odd assumption is that a project would follow the repo name. Many projects use different prefixes like `python-` in their name but not repo name. So the actual package name wouldn't be the same as the repo name, nor would it be the same as the main importable package name. Plus, there's src-layout where the importable package would be put under `src/` or `lib/` dir. And of course, there's a corner case with a Git repository having several independent projects in it.
FWIW, the project name could be sourced from these places:
* `setup.cfg` (sometimes even when there's no `setup.py`)
* `setup.py` (although, it's tricky to get it out of there)
* `pyproject.toml`
* there's also a multitude of superlegacy (hello, distutils) or weird ('ish, hello pbr!) mechanisms that I won't get into right now
In all cases, the version can be generated dynamically on build and depend on changes in Git (or other SCMs). For example, I tend to use `setuptools-scm` for this but there's more.
`setup.py`'s version can be dynamic even without any special plugins just because it's a Python module.
`setup.cfg` may contain a static version string but it can still be overridden by plugins like `setuptools-scm`. Also, there's a possibility to source it from an importable attribute (using the `attr:` syntax).
`pyproject.toml` may have a version declared as a dynamic field thanks to https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#version (formerly, [PEP 621](https://peps.python.org/pep-0621/) before the metadata moved to PyPUG with [PEP 643](https://peps.python.org/pep-0643/)).
The name field, OTOH, is forbidden to be specified dynamically with `setup.cfg` and `pyproject.toml` but because `setup.py` is a script, *theoretically* it could be dynamic (although I doubt anybody would want such a madness in practice).
In the modern world, package distributions are built via the PEP 517 interface. And using it, it should be possible to extract the version information from `.dist-info` / `.egg-info` metadata folders. PEP 517 specifies a [`prepare_metadata_for_build_wheel()`](https://peps.python.org/pep-0517/#prepare-metadata-for-build-wheel) and PEP 660 brings in [`prepare_metadata_for_build_editable()`](https://peps.python.org/pep-0660/#prepare-metadata-for-build-editable). The problem is that they are optional but the good news is that if you use PEP 517 against a legacy project, it'll automatically fall back to using setuptools (well, it's legacy backend to be precise).
So I'd recommend focusing on reading data from these dirs after calling the metadata preparation hooks. The metadata files should be readable with helpers from the `email` module IIRC.